### PR TITLE
added feat: bookmark add, edit, remove, view, save

### PR DESCRIPTION
**Pull request by themohammadsa**
# ✨ Pull Request
Add Bookmark Feature: Ability to Add, Remove, and View Bookmarks
---
### 📓 Referenced Issue
Fixes: #916
---

### ℹ️ About the PR
I have added the bookmark feature:
#### Bookmark Management
1. Adding Bookmarks:
- Users can add a bookmark by clicking on the star icon on the address bar.
- A flyout appears allowing users to edit the URL and name of the bookmark.
- Users can save the bookmark after making modifications.
2. Removing Bookmarks:
- Users can remove a bookmark using the same method.
- Clicking on the star icon on a bookmark opens the flyout.
- From the flyout, users can remove the bookmark.
3. Viewing Bookmarks:
- To view all bookmarks, users can click on the menu flyout button (vertical three dots) in the top-right corner.
- Clicking on the "Bookmarks" option in the dropdown menu displays a flyout with the list of bookmarks.
4. Empty Bookmarks:
- If no bookmarks are present, the UI text will indicate that there are no bookmarks.
- Users will be informed when there are no bookmarks available to display.

#### Editing Bookmarks
1. Modifying Bookmark Name and URL:
- Users can edit the bookmark's name and URL by clicking on the edit button.
- The edit button becomes visible when hovering over any bookmark item in the displayed bookmark list.
- Clicking the edit button opens a bookmark flyout edit.
- In the flyout, users can make modifications to the bookmark's name and URL.
2. Duplicate Bookmark Prevention:
- It is not possible to create duplicate copies of bookmarks.
- Each URL can have only one bookmark associated with it.
- However, users can have multiple bookmarks with the same name.
---
### 🔍 Testing Scenarios

I have tested the following scenarios:
1. Add Bookmark:
- Verify that clicking the star icon displays a flyout to edit the bookmark's URL and name.
- Save the bookmark and confirm it is added successfully.
2. Edit Bookmark:
- Hover over a bookmark item to show the edit button.
- Click the edit button and ensure the bookmark flyout edit appears.
- Modify the URL and name, save changes, and verify the bookmark is updated.
3. Remove Bookmark:
- Hover over a bookmark item to show the edit button.
- Click the edit button and confirm the bookmark flyout edit appears.
- Remove the bookmark and ensure it is successfully deleted.
4. Bookmark Icon Indicator:
- Check that the star icon on the address bar reflects bookmark status.
- Verify the icon changes appearance (filled/unfilled) based on bookmarked status.
- Add a bookmark and confirm the star icon reflects the bookmarked state.
5. View Bookmarks:
- Click the menu flyout button and select "Bookmarks."
- Verify the flyout displays the list of bookmarks correctly.
6. Duplicate Bookmark URLs:
- Test creating duplicate bookmarks for the same URL.
- Ensure only one bookmark is allowed per unique URL.
- Verify names can be repeated for different bookmarks.
7. Bookmark Persistence:
- Navigate through different pages and sections.
- Confirm bookmarks remain intact and accessible.
- Verify bookmarks persist even after closing and reopening the application.
8. No Bookmarks:
- Remove all bookmarks and check if the UI indicates no bookmarks.
- Verify appropriate messaging when no bookmarks are present.
---
### 🖼️ Screenshots
Screenshot 2023-06-10 at 1 46 11 PM
Screenshot 2023-06-10 at 1 45 33 PM
Screenshot 2023-06-10 at 1 45 45 PM
Screenshot 2023-06-10 at 1 46 41 PM
Screenshot 2023-06-10 at 1 47 03 PM

### CODE REVIEW

**Review by manojVivek — CHANGES_REQUESTED** [review]
@themohammadsa Great job on this one, added some minor comments (mostly nitpicks), otherwise looks good.
Just thrown the suggestions from the top of my head, please let me know if you have different thoughts and we can discuss.

But really awesome that you're pulling off huge features so quick!! 👏🏼

File: desktop-app/src/renderer/components/ToolBar/AddressBar/index.tsx [hunk 1]
```diff
@@ -240,6 +241,7 @@ const AddressBar = () => {
           >
             <Icon icon={isHomepage ? 'mdi:home' : 'mdi:home-outline'} />
           </Button>
+          <Bookmark currentAddress={address} />
```
[the remark this answers is not in the document]
Comment (themohammadsa) [reply desktop-app/src/renderer/components/ToolBar/AddressBar/index.tsx:244]: I can have a look at it for sure, 
but on second thoughts - I think it's a good idea to use this as an opportunity to make this into a ```good-first-issue```

File: desktop-app/src/store/index.ts [hunk 2]
```diff
@@ -190,6 +190,13 @@ const schema = {
     },
     default: [],
   },
+  bookmarks: {
+    type: 'array',
+    items: {
+      type: 'object',
```
Comment (manojVivek) [inline desktop-app/src/store/index.ts:196]: I think we can be a bit more specific here, to declare the `id`, `name` and `address` field as `properties` of the object.

**Review by manojVivek — APPROVED** [review]
Looking great now! Merging 🚀

**Pull request merged**